packages/roosterjs-editor-plugins/lib/plugins/ImageResize/ImageResizeHandler.ts

```diff
@@ -0,0 +1,266 @@
+            this.editor.addUndoSnapshot();
+            this.initWidth = this.image.clientWidth;
+            this.initHeight = this.image.clientHeight;
+            this.ratio =
+                this.initWidth > 0 && this.initHeight > 0
+                    ? (this.initWidth * 1.0) / this.initHeight
+                    : 0;
+        }
+    };
+
+    private onResize = ([x, y]: ResizeContext, deltaX: number, deltaY: number, e: MouseEvent) => {
+        if (this.image) {
+            const horizontalOnly = x == XCoordinate.Center;
+            const verticalOnly = y == YCoordinate.Middle;
+            const shouldPreserveRatio =
+                !(horizontalOnly || verticalOnly) && (this.forcePreserveRatio || e.shiftKey);
+            let newWidth = horizontalOnly
+                ? this.initWidth
+                : Math.max(
+                      this.initWidth + deltaX * (x == XCoordinate.Left ? -1 : 1),
+                      this.minWidth
+                  );
+            let newHeight = verticalOnly
+                ? this.initHeight
+                : Math.max(
+                      this.initHeight + deltaY * (y == YCoordinate.Top ? -1 : 1),
+                      this.minHeight
+                  );
+
+            if (shouldPreserveRatio) {
+                newHeight = Math.min(newHeight, (newWidth * this.initHeight) / this.initWidth);
+                newWidth = Math.min(newWidth, (newHeight * this.initWidth) / this.initHeight);
+
+                if (this.ratio > 0) {
+                    if (newWidth < newHeight * this.ratio) {
+                        newWidth = newHeight * this.ratio;
+                    } else {
+                        newHeight = newWidth / this.ratio;
+                    }
+                }
+            }
+
+            this.setSize(newWidth, newHeight);
+
+            // double check
+            if (shouldPreserveRatio) {
+                const clientWidth = Math.floor(this.image.clientWidth);
+                const clientHeight = Math.floor(this.image.clientHeight);
+                newWidth = Math.floor(newWidth);
+                newHeight = Math.floor(newHeight);
+                if (clientHeight !== newHeight || clientWidth !== newWidth) {
+                    if (clientHeight < newHeight) {
+                        newWidth = clientHeight * this.ratio;
+                    } else {
+                        newHeight = clientWidth / this.ratio;
+                    }
+
+                    this.setSize(newWidth, newHeight);
+                }
+            }
+        }
+    };
+
+    private onResizeEnd = () => {
+        if (this.image) {
+            this.editor.addUndoSnapshot();
+            this.editor.triggerContentChangedEvent(ChangeSource.ImageResize, this.image);
+        }
+    };
+
+    private setSize(width: number, height: number) {
+        this.image.style.width = width + 'px';
+        this.image.style.height = height + 'px';
+        this.image.width = width;
```

Comment:
I think style.width/height should be enough

Reply:
Outlook Desktop doesn't respect style.width/height, which causes resized image can't be shown in new size in Outlook Desktop. So that is necessary.

Reply:
okay. good to know that!